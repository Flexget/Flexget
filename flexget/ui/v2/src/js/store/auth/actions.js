const PREFIX = '@flexget/auth/';
export const LOGIN = `${PREFIX}LOGIN`;
export const LOGOUT = `${PREFIX}LOGOUT`;
