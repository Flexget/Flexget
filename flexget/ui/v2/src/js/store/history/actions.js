const PREFIX = '@flexget/history/';
export const GET_HISTORY = `${PREFIX}GET_HISTORY`;
