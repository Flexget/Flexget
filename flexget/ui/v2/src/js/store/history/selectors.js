import { createSelector } from 'reselect';
import { groupBy } from 'utils/array';

const getItems = history => history.items;
const getGrouping = (history, grouping) => grouping;
const historyGroupBy = (items, grouping, transform) => groupBy(items, grouping, transform);
const getHistoryGroupedByTime = createSelector(
  [getItems, getGrouping],
  (items, grouping) => historyGroupBy(items, grouping, i => i.split('T')[0]),
);
const getHistoryGroupedByTask = createSelector(
  [getItems, getGrouping],
  historyGroupBy,
);

export const getGroupedHistory = (history, grouping) => (
  grouping === 'time' ?
    getHistoryGroupedByTime :
    getHistoryGroupedByTask
)(history, grouping);
