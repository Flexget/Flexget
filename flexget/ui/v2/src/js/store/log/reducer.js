import {
  LOG_CONNECT,
  LOG_MESSAGE,
  LOG_DISCONNECT,
  LOG_CLEAR,
} from 'store/log/actions';

const initState = {
  messages: [],
  connected: false,
};

export default function reducer(state = initState, action) {
  switch (action.type) {
    case LOG_CONNECT:
      return {
        ...state,
        messages: [],
        connected: true,
      };
    case LOG_MESSAGE:
      return {
        ...state,
        messages: [action.payload, ...state.messages],
      };
    case LOG_DISCONNECT:
      return {
        ...state,
        connected: false,
      };
    case LOG_CLEAR:
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
}
