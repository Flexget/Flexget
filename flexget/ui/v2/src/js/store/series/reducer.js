import { combineReducers } from 'redux';
import shows from 'store/series/shows/reducer';

export default combineReducers({
  shows,
});
