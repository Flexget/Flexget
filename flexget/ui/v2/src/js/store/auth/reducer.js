import { LOGIN, LOGOUT } from 'store/auth/actions';
import { ERROR_STATUS } from 'store/status/actions';
import { GET_VERSION } from 'store/version/actions';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
    case GET_VERSION:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    case ERROR_STATUS:
      if (action.payload.statusCode === 401) {
        return {};
      }
    default: // eslint-disable-line no-fallthrough
      return state;
  }
}
