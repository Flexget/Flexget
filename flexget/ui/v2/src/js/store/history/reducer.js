import { GET_HISTORY } from 'store/history/actions';

const initState = {
  totalCount: 0,
  items: [],
};


export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_HISTORY:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.refresh ? payload.data : [...state.items, ...payload.data],
      };
    default:
      return state;
  }
}
