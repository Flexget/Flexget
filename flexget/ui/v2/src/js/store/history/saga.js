import { stringify } from 'qs';
import { call, put, takeLatest } from 'redux-saga/effects';
import { action, requesting } from 'utils/actions';
import { get } from 'utils/fetch';
import { GET_HISTORY } from 'store/history/actions';

export const defaultOptions = {
  page: 1,
  sort_by: 'time',
  order: 'desc',
};

export function* getHistory({ payload }) {
  const query = { ...defaultOptions, ...payload };
  const refresh = query.page === 1;

  try {
    const { data, headers } = yield call(get, `/history?${stringify(query)}`);
    yield put(action(GET_HISTORY, { data, headers, refresh }));
  } catch (err) {
    yield put(action(GET_HISTORY, err));
  }
}

export default function* saga() {
  yield takeLatest(requesting(GET_HISTORY), getHistory);
}
