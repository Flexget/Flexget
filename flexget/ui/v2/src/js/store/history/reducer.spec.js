import reducer from 'store/history/reducer';
import { GET_HISTORY } from 'store/history/actions';
import { Headers } from 'utils/tests';


describe('store/history/reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should store history on GET_HISTORY', () => {
    expect(reducer(undefined, {
      type: GET_HISTORY,
      payload: {
        data: [{ an: 'object' }],
        headers: new Headers({ 'total-count': 1 }),
      },
    })).toMatchSnapshot();
  });
});
