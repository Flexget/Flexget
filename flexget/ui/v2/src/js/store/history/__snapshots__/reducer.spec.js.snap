// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/history/reducer should return the initial state 1`] = `
Object {
  "items": Array [],
  "totalCount": 0,
}
`;

exports[`store/history/reducer should store history on GET_HISTORY 1`] = `
Object {
  "items": Array [
    Object {
      "an": "object",
    },
  ],
  "totalCount": 1,
}
`;
